arrow-select/src/take.rs

```diff
@@ -592,7 +592,10 @@ where
 
             let s = array.value(index);
 
-            length_so_far += T::Offset::from_usize(s.as_ref().len()).unwrap();
+            length_so_far += T::Offset::from_usize(
+                <<T as ByteArrayType>::Native as AsRef<[u8]>>::as_ref(s).len(),
+            )
+            .unwrap();
             values.extend_from_slice(s.as_ref());
```

Comment:
```suggestion
            ```suggestion
            let s: &[u8] = s.as_ref();
            length_so_far += T::Offset::from_usize(s.len()).unwrap();
            values.extend_from_slice(s);
```

Reply:
Oh Oh right. I followed the suggestion from the compiler and I did this later. 😅